java/feathub-connectors/flink-connectors/flink-connector-redis/src/main/java/com/alibaba/feathub/flink/connectors/redis/lookup/RedisLookupFunction.java

```diff
@@ -0,0 +1,191 @@
+ * redis key fields.
+ */
+public class RedisLookupFunction extends LookupFunction {
+
+    private final ReadableConfig config;
+
+    private final int[] originalKeyFieldIndices;
+    private final int[] redisKeyFieldIndices;
+    private final int[] redisValueFieldIndices;
+
+    private final DataType[] fieldTypes;
+
+    private transient JedisClient client;
+
+    private transient RowData.FieldGetter[] originalKeyGetters;
+
+    private transient RowData.FieldGetter[] redisKeyGetters;
+
+    public RedisLookupFunction(ReadableConfig config, ResolvedSchema schema) {
+        this.config = config;
+        this.originalKeyFieldIndices = getOriginalKeyFieldIndices(schema, config);
+        this.redisValueFieldIndices = getRedisValueFieldIndices(schema, config);
+        this.redisKeyFieldIndices = getRedisKeyFieldIndices(schema, redisValueFieldIndices);
+        this.fieldTypes = schema.getColumnDataTypes().toArray(new DataType[0]);
+    }
+
+    @Override
+    public void open(FunctionContext context) throws Exception {
+        super.open(context);
+        this.client = JedisClient.create(this.config);
+        this.originalKeyGetters = new RowData.FieldGetter[originalKeyFieldIndices.length];
+        for (int i = 0; i < originalKeyFieldIndices.length; i++) {
+            originalKeyGetters[i] =
+                    RowData.createFieldGetter(
+                            fieldTypes[originalKeyFieldIndices[i]].getLogicalType(), i);
+        }
+        this.redisKeyGetters = new RowData.FieldGetter[redisKeyFieldIndices.length];
+        for (int i = 0; i < redisKeyFieldIndices.length; i++) {
+            redisKeyGetters[i] =
+                    RowData.createFieldGetter(
+                            fieldTypes[redisKeyFieldIndices[i]].getLogicalType(),
+                            i + originalKeyFieldIndices.length);
+        }
+    }
+
+    @Override
+    public Collection<RowData> lookup(RowData rowData) throws IOException {
+        // TODO: Optimize performance when object reuse is enabled.
+        GenericRowData result = new GenericRowData(fieldTypes.length);
+
+        for (int i = 0; i < originalKeyFieldIndices.length; i++) {
+            result.setField(
+                    originalKeyFieldIndices[i], originalKeyGetters[i].getFieldOrNull(rowData));
+        }
+
+        for (int i = 0; i < redisValueFieldIndices.length; i++) {
+            result.setField(redisKeyFieldIndices[i], redisKeyGetters[i].getFieldOrNull(rowData));
+
+            String key =
+                    ConversionUtils.toString(
+                            rowData,
+                            i + originalKeyFieldIndices.length,
+                            fieldTypes[redisKeyFieldIndices[i]]);
+            int valueFieldIndex = redisValueFieldIndices[i];
+            DataType valueType = fieldTypes[valueFieldIndex];
+            if (valueType instanceof CollectionDataType) {
+                result.setField(
```

Comment:
This statement involves 3 function calls.

It might be more readable to separate it into 2-3 statements.

Same for the code below.